plugins/variational_quantum_classifier/tasks.py

```diff
@@ -0,0 +1,327 @@
+
+    if task_data is None:
+        msg = f"Could not load task data with id {db_id} to read parameters!"
+        TASK_LOGGER.error(msg)
+        raise KeyError(msg)
+
+    input_params: InputParameters = InputParametersSchema().loads(task_data.parameters)
+
+    train_data_url = input_params.train_data_url
+    train_labels_url = input_params.train_labels_url
+    test_data_url = input_params.test_data_url
+    test_labels_url = input_params.test_labels_url
+    feature_map_enum = input_params.feature_map
+    entanglement_pattern_feature_map_enum = input_params.entanglement_pattern_feature_map
+    reps_feature_map = input_params.reps_feature_map
+    paulis = input_params.paulis
+    vqc_ansatz_enum = input_params.vqc_ansatz
+    entanglement_pattern_ansatz_enum = input_params.entanglement_pattern_ansatz
+    reps_ansatz = input_params.reps_ansatz
+    optimizer_enum = input_params.optimizer
+    maxitr = input_params.maxitr
+    shots = input_params.shots
+    backend = input_params.backend
+    ibmq_token = input_params.ibmq_token
+    custom_backend = input_params.custom_backend
+    resolution = input_params.resolution
+
+    TASK_LOGGER.info(f"Loaded input parameters from db: {str(input_params)}")
+
+    if ibmq_token == "****":
+        TASK_LOGGER.info("Loading IBMQ token from environment variable")
+
+        if "IBMQ_TOKEN" in os.environ:
+            ibmq_token = os.environ["IBMQ_TOKEN"]
+            TASK_LOGGER.info("IBMQ token successfully loaded from environment variable")
+        else:
+            TASK_LOGGER.info("IBMQ_TOKEN environment variable not set")
+
+    # load data from file
+    train_id_to_idx, train_points = get_indices_and_point_arr(train_data_url)
+    train_labels = get_label_arr(train_labels_url, train_id_to_idx)
+
+    test_id_to_idx, test_points = get_indices_and_point_arr(test_data_url)
+    test_labels = None
+    if test_labels_url != "" and test_labels_url is not None:
+        test_labels = get_label_arr(test_labels_url, test_id_to_idx)
+
+    # set no. of qubits accordingly
+    n_qbits = train_points.shape[1]
+
+    # Prep backend
+    backend = backend.get_qiskit_backend(ibmq_token, custom_backend)
+    backend = QuantumInstance(backend=backend, shots=shots)
+
+    # Prep feature map
+    entanglement_pattern_feature_map = entanglement_pattern_feature_map_enum.get_pattern()
+    paulis = paulis.replace(" ", "").split(",")
+    feature_map = feature_map_enum.get_featuremap(
+        n_qbits, paulis, reps_feature_map, entanglement_pattern_feature_map
+    )
+
+    # Prep ansatz
+    entanglement_pattern_ansatz = entanglement_pattern_ansatz_enum.get_pattern()
+    vqc_ansatz = vqc_ansatz_enum.get_ansatz(
+        n_qbits, entanglement_pattern_ansatz, reps_ansatz
+    )
+
+    # Prep optimizer
+    optimizer = optimizer_enum.get_optimizer(maxitr)
+
+    # VQC
+    vqc = qiskitVQC(backend, feature_map, vqc_ansatz, optimizer)
+    vqc.fit(train_points, train_labels)
```

Comment:
It would be nice to display the progress of the training, because it can take a while.

Reply:
I couldn't realy make out a solution on how to display the training progress. There is an option for a so called "callback" function. The callback function is called every time the objective function is called. The optimizer might use the objective function multiple times during one iteration, thus I cannot trake the current iteration.
Also, when using the adam optimizer, my callback function gets called only once, after training. This is very unusual and I am not sure what to make of this.

Do you have a solution? If not, I can look further into this.

Reply:
I don't know a better solution than using the callback function and since the callback function seems to be unreliable, I would say that you don't need to put more time into this.